### DISCUSSION TITLE
`settings.<name>.custom` perhaps not working as documented

### DISCUSSION THREAD

**Discussion by raehik** [srid/haskell-flake#201] [Q&A] [answered]
I need to override a package with a newer one. I could do so with `packages.<name>.source`, but it needs more Nix-specific configuration, and it turns out nixpkgs has the newer one with all that config already done.

The documentation https://zero-to-flakes.com/haskell-flake/dependency#nixpkgs says I can do this:

```nix
haskellProjects.default = {
  settings = { super, ... }: {
    fourmolu.custom = super.fourmolu_0_13_1_0;
  };
};
```

I use that syntax, but receive the error `A definition for option 'custom' is not of type 'null or (function that evaluates to a(n) package)'`. PR here: https://github.com/raehik/generic-data-functions/pull/3 , package attempting to override is `splitmix`.

I've found relevant code, but I'm not sure how to understand it: https://github.com/srid/haskell-flake/blob/63a160879e6700b1c9385852043b162183063e9a/nix/modules/project/settings/all.nix#L307-L323

**Comment by srid** [comment, marked as answer]
It needs to be a function. See example at https://github.com/srid/haskell-flake/issues/192 where we also discuss the possibility of improving this interface.

```nix
haskellProjects.default = {
  settings = {
    fourmolu = { super, ...}: { custom = _: super.fourmolu_0_13_1_0; };
  };
};
```

**Reply by raehik** [reply]
Thanks as always! I think the docs on zero-to-flakes wants an update to reflect that syntax.

**Reply by srid** [reply]
It is briefly mentioned as an example in the docs:

https://community.flake.parts/haskell-flake/dependency#nixpkgs
